solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -0,0 +1,369 @@
+#![allow(dead_code)]
+
+use super::single_pair_settlement::{AmmSwapExactTokensForTokens, SinglePairSettlement};
+use crate::settlement::Trade;
+use anyhow::{anyhow, Result};
+use model::order::{OrderCreation, OrderKind};
+use num::{bigint::Sign, BigInt};
+use std::collections::HashMap;
+use web3::types::{Address, U256};
+
+#[derive(Default)]
+struct TokenContext {
+    address: Address,
+    reserve: U256,
+    buy_volume: U256,
+    sell_volume: U256,
+}
+
+impl TokenContext {
+    pub fn is_excess(&self, other: &TokenContext) -> bool {
+        u256_to_bigint(&self.reserve)
+            * (u256_to_bigint(&other.sell_volume) - u256_to_bigint(&other.buy_volume))
+            < u256_to_bigint(&other.reserve)
+                * (u256_to_bigint(&self.sell_volume) - u256_to_bigint(&self.buy_volume))
+    }
```

Comment:
I think this is (5) from the doc but in case isn't the fee missing?

Reply:
Is this `u256_to_bigint` something that could be upstreamed to primitive types? Looks a bit ugly here.

Reply:
For the fee, see below. I'm assuming 0 protocol fee for now (even when introduced my assumption is we can deduct the fee before we pass in the orders). 

> Is this u256_to_bigint something that could be upstreamed to primitive types?

I'm not sure bigint is common enough to justify such a conversion (before we were doing it with std types but here we are dealing with two separate crates).